### DISCUSSION TITLE
SOFT FAIL SSL delivering mail

### DISCUSSION THREAD

**Discussion by fernandoSynetic** [postalserver/postal#2690] [Help with using Postal]
Hi there,

I have a problem with delivery to several mailboxes and I don't know how to solve it. Apparently it may come from the self-signed certificate, but I'm not sure. It works correctly on other servers.

These are the logs. Thank you very much in advance.

worker_1    | [9] [2023-11-09T12:15:26.500] INFO -- : [AWWNFA15OL] Lock acquired for queued message 318
worker_1    | [9] [2023-11-09T12:15:26.505] INFO -- : [AWWNFA15OL] Found 0 associated messages to process at the same time (batch key: outgoing-q8spain.es)
worker_1    | [9] [2023-11-09T12:15:26.506] INFO -- : [AWWNFA15OL] [1::317 318] Got queued message with exclusive lock
worker_1    | [9] [2023-11-09T12:15:45.277] INFO -- : [RKIV4LRV] Cannot connect to 88.221.213.113:25 (mail.q8.it) (IO::TimeoutError: connect(2) for "88.221.213.113" port 25)
worker_1    | [9] [2023-11-09T12:15:45.278] INFO -- : [VZV5TZTD] Cannot connect to 88.221.213.113:25 (mail.q8.it) (IO::TimeoutError: connect(2) for "88.221.213.113" port 25)
worker_1    | [9] [2023-11-09T12:15:45.278] INFO -- : [SUJZJAPC] Cannot connect to 88.221.213.113:25 (mail.q8.it) (IO::TimeoutError: connect(2) for "88.221.213.113" port 25)
worker_1    | [9] [2023-11-09T12:15:45.554] INFO -- : [RKIV4LRV] SSL error (SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: dh key too small), retrying without SSL
worker_1    | [9] [2023-11-09T12:15:45.594] INFO -- : [VZV5TZTD] SSL error (SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: dh key too small), retrying without SSL
worker_1    | [9] [2023-11-09T12:15:45.608] INFO -- : [SUJZJAPC] SSL error (SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: dh key too small), retrying without SSL
worker_1    | [9] [2023-11-09T12:15:45.829] INFO -- : [VZV5TZTD] Cannot connect to 94.95.109.72:25 (mailbk.q8.it) (OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: certificate verify failed (self signed certificate))
worker_1    | [9] [2023-11-09T12:15:45.829] INFO -- : [RKIV4LRV] Cannot connect to 94.95.109.72:25 (mailbk.q8.it) (OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: certificate verify failed (self signed certificate))
worker_1    | [9] [2023-11-09T12:15:45.838] INFO -- : [SUJZJAPC] Cannot connect to 94.95.109.72:25 (mailbk.q8.it) (OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: certificate verify failed (self signed certificate))
worker_1    | [9] [2023-11-09T12:15:46.122] INFO -- : [RKIV4LRV] SSL error (SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: dh key too small), retrying without SSL
worker_1    | [9] [2023-11-09T12:15:46.136] INFO -- : [SUJZJAPC] SSL error (SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: dh key too small), retrying without SSL
worker_1    | [9] [2023-11-09T12:15:46.390] INFO -- : [SUJZJAPC] Cannot connect to 88.53.116.245:25 (mailna.q8.it) (OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: certificate verify failed (self signed certificate))
worker_1    | [9] [2023-11-09T12:15:46.394] INFO -- : [SUJZJAPC] -> No SMTP server available for q8spain.es
worker_1    | [9] [2023-11-09T12:15:46.394] INFO -- : [SUJZJAPC] -> Hostnames: ["mail.q8.it", "mailbk.q8.it", "mailna.q8.it"]
worker_1    | [9] [2023-11-09T12:15:46.395] INFO -- : [SUJZJAPC] -> Errors: ["connect(2) for \"88.221.213.113\" port 25", "SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: certificate verify failed (self signed certificate)", "SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: certificate verify failed (self signed certificate)"]
worker_1    | [9] [2023-11-09T12:15:46.408] INFO -- : [Q3PACZDBEQ] [1::314 315] Message requeued for trying later.
worker_1    | [9] [2023-11-09T12:15:46.414] INFO -- : [SUJZJAPC] Finishing up
worker_1    | [9] [2023-11-09T12:15:46.517] INFO -- : [Q3PACZDBEQ] Finished processing UnqueueMessageJob job in 130.790625462s
worker_1    | [9] [2023-11-09T12:15:46.518] INFO -- : [VZV5TZTD] SSL error (SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: dh key too small), retrying without SSL
worker_1    | [9] [2023-11-09T12:15:46.806] INFO -- : [VZV5TZTD] Cannot connect to 88.53.116.245:25 (mailna.q8.it) (OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: certificate verify failed (self signed certificate))
worker_1    | [9] [2023-11-09T12:15:46.810] INFO -- : [VZV5TZTD] -> No SMTP server available for q8spain.es
worker_1    | [9] [2023-11-09T12:15:46.810] INFO -- : [VZV5TZTD] -> Hostnames: ["mail.q8.it", "mailbk.q8.it", "mailna.q8.it"]
worker_1    | [9] [2023-11-09T12:15:46.810] INFO -- : [VZV5TZTD] -> Errors: ["connect(2) for \"88.221.213.113\" port 25", "SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: certificate verify failed (self signed certificate)", "SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: certificate verify failed (self signed certificate)"]
worker_1    | [9] [2023-11-09T12:15:46.818] INFO -- : [QVOOGDVX3H] [1::315 316] Message requeued for trying later.
worker_1    | [9] [2023-11-09T12:15:46.821] INFO -- : [VZV5TZTD] Finishing up
worker_1    | [9] [2023-11-09T12:15:46.881] INFO -- : [QVOOGDVX3H] Finished processing UnqueueMessageJob job in 131.154231802s
worker_1    | [9] [2023-11-09T12:15:47.778] INFO -- : [RKIV4LRV] Cannot connect to 88.53.116.245:25 (mailna.q8.it) (OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: certificate verify failed (self signed certificate))
worker_1    | [9] [2023-11-09T12:15:47.781] INFO -- : [RKIV4LRV] -> No SMTP server available for q8spain.es
worker_1    | [9] [2023-11-09T12:15:47.781] INFO -- : [RKIV4LRV] -> Hostnames: ["mail.q8.it", "mailbk.q8.it", "mailna.q8.it"]
worker_1    | [9] [2023-11-09T12:15:47.781] INFO -- : [RKIV4LRV] -> Errors: ["connect(2) for \"88.221.213.113\" port 25", "SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: certificate verify failed (self signed certificate)", "SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: certificate verify failed (self signed certificate)"]
worker_1    | [9] [2023-11-09T12:15:47.816] INFO -- : [FCQBOT24OP] [1::316 317] Message requeued for trying later.

**Comment by willpower232** [comment]
Have you provided certificates and changed the setting to true?

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/config/postal.defaults.yml#L69-L71

**Comment by fernandoSynetic** [comment]
thanks for your answer.

No settings have been changed.
Other emails are sent without problem but it fails on that server.

**Reply by fernandoSynetic** [reply]
Thank you for your help. I have enabled TLS on the server, but I am still encountering errors when delivering mail to that domain. It seems to be using a self-signed certificate, and I don't know how to resolve it.

`connect(2) for "88.221.213.81" port 25, SSL_connect returned=1 errno=0 peeraddr=94.95.109.72:25 state=error: certificate verify failed (self signed certificate), SSL_connect returned=1 errno=0 peeraddr=88.53.116.245:25 state=error: certificate verify failed (self signed certificate) Support Ref: AUPLVD0G`

I don't know if there is any solution from Postal. The options I read about involve lowering the overall security, and I'm not comfortable with that.

Thanks.

**Reply by willpower232** [reply]
Have you created certificates and set the paths in the config file?

**Reply by fernandoSynetic** [reply]
Yes, this is my postal.yml:

`smtp_server:
  # Specify configuration to the Postal SMTP server
  port: 25
  tls_enabled: true
  tls_certificate_path: /config/smtp.crt
  tls_private_key_path: /config/smtp.key`

I tested OK in https://www.gmass.co/smtp-test with TLS.

Any idea?
Thanks!

**Reply by vaztimur** [reply]
same issue. DId everything like here and in other topics described. GMASS shows an error

**Comment by fernandoSynetic** [comment]
I have a question about the TLS activation process in Postal. 

I have verified that now if I connect to my server through port 25 TLS is activated. But I think what I really need to avoid these errors is for Postal to use TLS when connecting to other servers to deliver mail. 

I'm wrong? How can I check that TLS is used upon delivery?
